drv/lpc55-spi-server/src/main.rs

```diff
@@ -160,66 +143,10 @@ fn turn_on_flexcomm(syscon: &Syscon) {
     syscon.leave_reset(Peripheral::Fc3).unwrap_lite();
 }
 
-fn muck_with_gpios(syscon: &Syscon) {
-    syscon.enable_clock(Peripheral::Iocon).unwrap_lite();
-    syscon.leave_reset(Peripheral::Iocon).unwrap_lite();
-
+fn muck_with_gpios() {
     let gpio_driver = GPIO.get_task_id();
-    let iocon = Pins::from(gpio_driver);
-
-    // This is quite the array!
-    // All of these need to be in digital mode. The NXP C driver
-    // also sets the pull-up resistor
-
-    let pin_settings = [
-        // HS_SPI_MOSI = P0_26 = FUN9
-        (
-            Pin::PIO0_26,
-            AltFn::Alt9,
-            Mode::PullUp,
-            Slew::Standard,
-            Invert::Disable,
-            Digimode::Digital,
-            Opendrain::Normal,
-        ),
-        // HS_SPI_MISO = P1_3 = FUN6
-        (
-            Pin::PIO1_3,
-            AltFn::Alt6,
-            Mode::PullUp,
-            Slew::Standard,
-            Invert::Disable,
-            Digimode::Digital,
-            Opendrain::Normal,
-        ),
-        // HS_SPI_SCK = P1_2 = FUN6
-        (
-            Pin::PIO1_2,
-            AltFn::Alt6,
-            Mode::PullUp,
-            Slew::Standard,
-            Invert::Disable,
-            Digimode::Digital,
-            Opendrain::Normal,
-        ),
-        // HS_SPI_SSEL1 = P1_1 = FUN5
-        // Note that SSEL0, SSEL2 and SSEL3 are not used in the current design
-        (
-            Pin::PIO1_1,
-            AltFn::Alt5,
-            Mode::PullUp,
-            Slew::Standard,
-            Invert::Disable,
-            Digimode::Digital,
-            Opendrain::Normal,
-        ),
-    ];
-
-    for (pin, alt, mode, slew, invert, digi, od) in
-        core::array::IntoIter::new(pin_settings)
-    {
-        iocon
-            .iocon_configure(pin, alt, mode, slew, invert, digi, od)
-            .unwrap_lite();
-    }
+
+    setup_pins(gpio_driver).unwrap_lite();
```

Comment:
At this point, might as well delete `muck_with_gpios()` entirely and just put these two lines into `main`?